src/lib/TranscriptEditor/TimedTextEditor/index.js

```diff
@@ -222,6 +226,104 @@ class TimedTextEditor extends React.Component {
     return currentWord;
   }
 
+  myKeyBindingFn = ( e) => {
+    const enterKey = 13;
+    if (e.keyCode === enterKey ) {
+      return 'split-paragraph';
+    }
+    return getDefaultKeyBinding(e);
+  }
+
+  // TODO: code in this function can do with a refactor - at later stage  
+  handleKeyCommand = (command) => {
+    // https://github.com/facebook/draft-js/issues/723#issuecomment-367918580
+    // https://draftjs.org/docs/api-reference-selection-state#start-end-vs-anchor-focus
+    if (command === 'split-paragraph') {
+      // on enter key, perform split paragraph at selection point 
+      const currentSelection = this.state.editorState.getSelection();
+      
+      if (currentSelection.isCollapsed()) {
+        const currentContent = this.state.editorState.getCurrentContent();
+        // https://draftjs.org/docs/api-reference-modifier#splitblock
+        const newContentState = Modifier.splitBlock(currentContent, currentSelection)
+        // https://draftjs.org/docs/api-reference-editor-state#push
+        const splitState = EditorState.push(this.state.editorState, newContentState, 'split-block')
+        const targetSelection = splitState.getSelection();
+        const originalBlock = currentContent.blockMap.get(newContentState.selectionBefore.getStartKey());
+        const originalBlockData = originalBlock.getData();
+        const blockSpeaker = originalBlockData.get('speaker');
+        // TODO: there might be some edge cases where unable to calculate wordStartTime
+        // eg adding spaces and then new line in the middle
+        let wordStartTime = 'NA';
+        let isEndOfParagraph = false;
+
+        let entityKey = originalBlock.getEntityAt(currentSelection.getStartOffset());
+        const startSelectionOffsetKey = currentSelection.getStartOffset();
+        // length of the plaintext for the ContentBlock
+        const lengthPlainTextForTheBlock = originalBlock.getLength()
+        // number of char from selection point to end of paragraph 
+        const remainingCharNumber = lengthPlainTextForTheBlock - startSelectionOffsetKey;
+        // if there is no word entity associated with char
+        if(entityKey === null){
```

Comment:
Just formatting issues again - did your IDE sync up with the linter? I'm not making a big fuss but stuff like spacing around brackets (particularly the few lines below this), semi-colons etc. is kinda messy.

Reply:
yeah not sure 🤷‍♂️

Reply:
dunno how to check this, but will need to look it to it at some point